### PR TITLE
Schema augmentation function

### PR DESCRIPTION
**Pull request by alexivkin**
Added a way to augment existing schema, instead of generating one from the scratch. This is needed for importing multiple sets of data, where each only has a subset of all of the possible fields. Even if none of the data sets have complete list of fields, this option allows incremental expansion of the schema.

It is also helpful when the schema is hand-edited and you do not want to lose changes.

### CODE REVIEW

**Comment by abroglesc** [comment]
@alexivkin I believe shares a lot of overlap with #57 which seems to cover more corner cases that could occur.

**Comment by alexivkin** [comment]
yup, seems to do the same stuff. I wish you actually merged that request before I had to whip out one myself.

**Comment by alexivkin** [comment]
Unrelated to the existing schema - there is still a bug in the original with integers in scientific notation appearing as floats.

File: bigquery_schema_generator/generate_schema.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ class SchemaGenerator:
     TIME_MATCHER = re.compile(r'^\d{1,2}:\d{1,2}:\d{1,2}(\.\d{1,6})?$')
 
     # Detect integers inside quotes.
-    INTEGER_MATCHER = re.compile(r'^[-]?\d+$')
+    INTEGER_MATCHER = re.compile(r'^-?\d+\.?(?:\d+[e|E]\+\d+)?$') # match scientific notation as well as (^-?\d+$)
```
Comment (bxparks) [inline bigquery_schema_generator/generate_schema.py]: Please send this change as a separate PR.

**Review by bxparks — COMMENTED** [review]
The problem with floating point is a separate issue, and should be sent as a separate PR. But let's wait until #57 is merged first.

File: bigquery_schema_generator/generate_schema.py (hunk 1 above)
Comment (bxparks) [reply bigquery_schema_generator/generate_schema.py]: #57 has been merged as #62. Please rebase on to `develop` and update this PR.
With regards to the floating point in quotes, I don't think I understand the problem.

Comment (alexivkin) [reply bigquery_schema_generator/generate_schema.py]: Python's JSON parser comprehends scientific notation but thinks that it must be a float, even for the integers like "1.43e5", hence the forced is_integer() check.

**Comment by ZiggerZZ** [comment]
It's a nice feature to have!
Is this library the right place to implement a function that checks if one BQ schema is a subset of another? Personally, I need it when APIs that I use add new fields to the data.

File: bigquery_schema_generator/generate_schema.py (hunk 1 above)
Comment (bxparks) [reply bigquery_schema_generator/generate_schema.py]: I see. I think you correctly point out that `INTEGER_MATCHER` and `FLOAT_MATCHER` do not accept all forms of integers and floats that `bq load` accepts. However, I think you are mistaken about "1.43e5" being an integer, because `bq load --autodetect` infers that type as a float.

I submitted a commit that updates `INTEGER_MATCHER` and `FLOAT_MATCHER` to be the following:

```python
INTEGER_MATCHER = re.compile(r'^[-+]?\d+$')
FLOAT_MATCHER = re.compile(r'^[-+]?(?:\d+\.?\d*|\.\d+)(?:[eE][-+]?\d+)?$') 
```

I added unit tests, and ran a bunch of experiments of `bq load --autodtect` and these seem to match the behavior of `bq load`.

Maybe you can pull the latest version from `develop` and let me know if it works for you.

Comment (bxparks) [reply bigquery_schema_generator/generate_schema.py]: I released these changes as v1.4 and pushed to PyPI. I am inclined to close this PR.

**Comment by bxparks** [comment]
I think this PR is obsolete. Closing.
